fast64_internal/oot/oot_utility.py

```diff
@@ -602,3 +633,282 @@ def oot_utility_register():
+
+		if actorType is not None:
+			if len(tiedList) == 1 and tiedList[0] == actorType:
+				return True
+			else:
+				for j in range(len(tiedList)):
+					if tiedList[j] == actorType:
+						return True
+	return False
+
+def getActorString(elem, detailedProp, field, lenProp, lenSwitch, lenBool, lenEnum, paramTarget):
+	'''Returns the current actor's parameter string'''
+	shiftTmp = 0
+	params = tmpParams = ""
+	charOR = " | "
+	strMask = elem.get('Mask')
+	target = elem.get('Target')
+	if target is None: target = "Params"
+	if elem.tag != 'Parameter' and strMask is not None:
+		mask = int(strMask, base=16)
+		shiftTmp = len(f'{mask:016b}') - len(f'{mask:016b}'.rstrip('0'))
+		shift = f'{shiftTmp}'
+		if elem.tag == 'Flag':
+			elemType = elem.get('Type')
+			if elemType == 'Chest' and target == paramTarget:
+				tmpParams = getActorParameter(detailedProp, field + '.chestFlag', shift, strMask)
+				if tmpParams != "":
+					params += tmpParams + charOR
+			elif elemType == 'Collectible' and target == paramTarget:
+				tmpParams = getActorParameter(detailedProp, field + '.collectibleFlag', shift, strMask)
+				if tmpParams != "":
+					params += tmpParams + charOR
+			elif elemType == 'Switch' and target == paramTarget:
+				for i in range(1, (int(lenSwitch, base=10) + 1)):
+					if i == int(elem.get('Index'), base=10) and target == paramTarget:
+						tmpParams = getActorParameter(detailedProp, field + f'.switchFlag{i}', shift, strMask)
+						if tmpParams != "":
+							params += tmpParams + charOR
+		elif elem.tag == 'Property' and elem.get('Name') != 'None':
+			for i in range(1, (int(lenProp, base=10) + 1)):
+				if i == int(elem.get('Index'), base=10) and target == paramTarget:
+					tmpParams = getActorParameter(detailedProp, (field + f'.props{i}'), shift, strMask)
+					if tmpParams != "":
+						params += tmpParams + charOR
+		elif elem.tag == 'ChestContent' and target == paramTarget:
+			if shift != '0':
+				params += f"(({detailedProp.itemChest} << {shift}) & 0x{mask:X}) | "
+			else:
+				params += f"({detailedProp.itemChest} & 0x{mask:X}) | "
+		elif elem.tag == 'Message' and target == paramTarget:
+			if shift != '0':
+				params += f"(({detailedProp.naviMsgID} << {shift}) & 0x{mask:X}) | "
+			else:
+				params += f"({detailedProp.naviMsgID} & 0x{mask:X}) | "
+		elif elem.tag == 'Collectible' and target == paramTarget:
+			tmpParams = getActorParameter(detailedProp, field + '.collectibleDrop', shift, strMask)
+			if tmpParams != "":
+				params += tmpParams + charOR
+		elif elem.tag == 'Bool':
+			for i in range(1, (int(lenBool, base=10) + 1)):
+				if i == int(elem.get('Index'), base=10) and target == paramTarget:
+					tmpParams = getActorParameter(detailedProp, (field + f'.bool{i}'), shift, strMask)
+					if tmpParams != "":
+						params += tmpParams + charOR
+		elif elem.tag == 'Enum':
+			for i in range(1, (int(lenEnum, base=10) + 1)):
+				if i == int(elem.get('Index'), base=10) and target == paramTarget:
```

Comment:
is this different from
```suggestion
			i = int(elem.get('Index'), base=10)
			if i >= 1 and i <= int(lenEnum, base=10):
				if target == paramTarget:
```
🤔 ?